chapters/05-raster-vector.qmd

```diff
@@ -1,7 +1,797 @@
+
+```{julia}
+zion_transect_pnt = GO.reproject(GI.getpoint(zion_transect_line); target_crs = GI.crs(src_srtm), source_crs = GI.crs(zion_transect_line))
+```
+
+
+Finally, we extract the elevation values for each point in our transect and combine the information with `zion_transect_pnt` (after "promoting" it to a `GeoDataFrame`, to accommodate extra attributes), using the point extraction method shown earlier (@sec-extraction-to-points).
+We also attach the respective distance cutoff points `distances`.
+
+```{julia}
+zion_transect_pnt = DataFrame(geometry = zion_transect_pnt)
+
+result = Rasters.extract(src_srtm, zion_transect_pnt; geometry = false) .|> first
+zion_transect_pnt[!, "elev"] = result
+```
+
+We also want to visualize an elevation profile along this line, so we compute the distances along the line manually. Using `GO.distance`, we can get the distance between successive points along the line.  Then, we add `0.0` to the beginning of the array of distances, and sum along that array to get the cumulative distance along the line.
+
+```{julia}
+# Compute distances between successive points along the line
+distances_between_points = GO.distance.(zion_transect_pnt.geometry[1:end-1], zion_transect_pnt.geometry[2:end])
+# Compute cumulative distances along the line
+cumulative_distances = cumsum([0.0; distances_between_points])
+zion_transect_pnt[!, "dist"] = cumulative_distances
+zion_transect_pnt
+```
+
+
+The information in `zion_transect_pnt`, namely the `"dist"` and `"elev"` attributes, can now be used to draw an elevation profile, as illustrated in @fig-zion-transect.
+
+
+```{julia}
+#| label: fig-zion-transect
+#| fig-cap: Extracting a raster values profile to line 
+#| layout-ncol: 2
+#| fig-subcap: 
+#| - Raster and a line transect
+#| - Extracted elevation profile
+# Raster and a line transect
+fig, ax, plt = plot(src_srtm)
+lines!(ax, zion_transect; color = :black)
+poly!(ax, zion.geom; color = :transparent, strokecolor = :white, strokewidth = 0.75)
+display(fig)
+# Elevation profile
+fig, ax, plt = lines(
+    zion_transect_pnt.dist, 
+    zion_transect_pnt.elev;
+    axis = (;
+        xlabel = "Distance (m)",
+        ylabel = "Elevation (m)",
+    )
+)
+fig
+```
+
+### Extraction to polygons {#sec-extraction-to-polygons}
+
+The final type of geographic vector object for raster extraction is polygons.
+Like lines, polygons tend to return many raster values per vector geometry.
+For continuous rasters (@fig-raster-extract-to-polygon (a)), we typically want to generate summary statistics for raster values per polygon, for example to characterize a single region or to compare many regions.
+The generation of raster summary statistics, by polygons, is demonstrated in the code below using `Rasters.zonal`, which creates a list of summary statistics (in this case a list of length 1, since there is just one polygon).
```

Comment:
```suggestion
The generation of raster summary statistics, by polygons, is demonstrated in the code below using `Rasters.zonal`, which creates a list of summary statistics for "zones" defined by geometry.
In this case, a vector of length 1 is returned, since there is just one polygon in the `DataFrame`.
```